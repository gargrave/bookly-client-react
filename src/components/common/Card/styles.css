.bookly-card {
  background-color: #f9f9f9;
  cursor: pointer;
  border: solid 1px #bbb;
  border-radius: 5px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  margin: 10px auto;
  width: 95%;
  max-width: 600px;
  padding: 8px;
}

.bookly-card:hover {
  background: #ffe8e4;
}

.bookly-card p {
  margin: 0;
}

.bookly-card .bookly-card-title {
  font-weight: bold;
}

.bookly-card .bookly-card-text {
  color: #6e6969;
}
