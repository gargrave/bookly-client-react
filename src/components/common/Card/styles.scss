.bookly-card {
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  box-shadow:
    0 1px 3px rgba(0, 0, 0, 0.07),
    0 1px 2px rgba(0, 0, 0, 0.12);
  margin: 8px 10px;
  padding: 10px 20px;
  width: 95%;

  hr {
    margin: 12px 0;
  }

  &.bookly-hoverable:hover {
    background: #ffe8e4;
    cursor: pointer;
  }

  p {
    margin: 0;
  }

  &-title {
    font-weight: bold;
  }

  &-text {
    color: #6e6969;
  }

  &-text + &-text {
    margin-top: 8px;
  }

  &-spacer {
    height: 20px;
    line-height: 20px;
  }

  &.bookly-detail-card {
    .bookly-card-title {
      font-size: 1.3rem;
      line-height: 2.5rem;
    }

    .bookly-card-title + .bookly-card-text {
      font-size: 1.1rem;
    }
  }
}
