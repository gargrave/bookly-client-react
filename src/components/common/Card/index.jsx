import React from 'react'
import { array, func, string } from 'prop-types'

import { buildClasses } from '@/utils/cssHelpers'

import './styles.css'

const renderText = (text, classname) => (
  text
    ? <p className={buildClasses(classname)}>{text}</p>
    : null
)

const Card = ({
  children,
  classes = [],
  onClick,
  text,
  title,
}) => {
  return (
    <div
      className={buildClasses(['card', ...classes])}
      onClick={onClick}>
      {renderText(title, 'card-title')}
      {renderText(text, 'card-text')}
      {children}
    </div>
  )
}

Card.propTypes = {
  children: array,
  classes: array,
  onClick: func.isRequired,
  text: string,
  title: string,
}

export default Card
