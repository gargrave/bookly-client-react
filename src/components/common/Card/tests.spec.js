import React from 'react'
import { shallow } from 'enzyme'

import Card from './'

describe('Card', () => {
  let component
  let props

  beforeEach(() => {
    props = {
      classes: [],
      onClick: jest.fn(),
      text: 'this is the text',
      title: 'Awesome Title',
    }
  })

  it('renders correctly', () => {
    component = shallow(<Card {...props} />)
    expect(component).toMatchSnapshot()
    expect(component.find('.bookly-card').length).toEqual(1)
    expect(component.find('.bookly-card-title').length).toEqual(1)
    expect(component.find('.bookly-card-text').length).toEqual(1)
  })

  it('does not render title when prop is blank', () => {
    props.title = ''
    component = shallow(<Card {...props} />)
    expect(component.find('.bookly-card-title').length).toEqual(0)
  })

  it('does not render text when prop is blank', () => {
    props.text = ''
    component = shallow(<Card {...props} />)
    expect(component.find('.bookly-card-text').length).toEqual(0)
  })

  it('adds extra classes correctly', () => {
    props.classes = ['oneclass', 'anotherclass']
    component = shallow(<Card {...props} />)
    expect(component.find('.bookly-oneclass.bookly-anotherclass').length).toEqual(1)
  })

  it('calls the "onClick" callback when clicked', () => {
    component = shallow(<Card {...props} />)
    expect(props.onClick.mock.calls.length).toBe(0)
    component.simulate('click')
    expect(props.onClick.mock.calls.length).toBe(1)
  })
})
