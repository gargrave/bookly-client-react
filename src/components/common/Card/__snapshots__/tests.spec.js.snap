// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="bookly-card"
    onClick={[Function]}
>
    <p
        className="bookly-card-title"
    >
        Awesome Title
    </p>
    <p
        className="bookly-card-text"
    >
        this is the text
    </p>
</div>,
  "nodes": Array [
    <div
      className="bookly-card"
      onClick={[Function]}
>
      <p
            className="bookly-card-title"
      >
            Awesome Title
      </p>
      <p
            className="bookly-card-text"
      >
            this is the text
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Card
        classes={Array []}
        onClick={[Function]}
        text="this is the text"
        title="Awesome Title"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "classes": Array [],
          "onClick": [Function],
          "text": "this is the text",
          "title": "Awesome Title",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="bookly-card"
          onClick={[Function]}
>
          <p
                    className="bookly-card-title"
          >
                    Awesome Title
          </p>
          <p
                    className="bookly-card-text"
          >
                    this is the text
          </p>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="bookly-card"
          onClick={[Function]}
>
          <p
                    className="bookly-card-title"
          >
                    Awesome Title
          </p>
          <p
                    className="bookly-card-text"
          >
                    this is the text
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Card
    classes={Array []}
    onClick={[Function]}
    text="this is the text"
    title="Awesome Title"
/>,
}
`;
